docs/sphinx/user_guide/feature/reduction.rst

```diff
@@ -236,39 +240,74 @@ RAJA::expt::Reduce
           compatible with the ``EXEC_POL``. ``Seg`` is the iteration space
           object for ``RAJA::forall``.
 
-.. important:: The order and types of the local reduction variables in the
-               kernel body lambda expression must match exactly with the
-               corresponding ``RAJA::expt::Reduce`` arguments to the
-               ``RAJA::forall`` to ensure that the correct result is obtained.
+.. important:: The order of the local reduction variables in the
+               kernel body lambda expression must ``RAJA::expt::ValOp`` objects which
+               match exactly in the underlying reduction data type, and RAJA
+               operator, with the corresponding ``RAJA::expt::Reduce`` arguments
+               to the ``RAJA::forall`` to ensure that the correct result is obtained.
 
 RAJA::expt::ValLoc
 ..................
 
 As with the current RAJA reduction interface, the new interface supports *loc*
 reductions, which provide the ability to get a kernel/loop index at which the
 final reduction value was found. With this new interface, *loc* reductions
-are performed using ``ValLoc<T>`` types. Since they are strongly typed, they
-provide ``min()`` and ``max()`` operations that are equivalent to using
-``RAJA_MIN()`` or ``RAJA_MAX`` macros as demonstrated in the code example below.
-Users must use the ``getVal()`` and ``getLoc()`` methods to access the reduction
-results::
+are performed using ``ValLoc<T,I>`` types, where ``T`` is the underlying data type,
+and ``I`` is the index type. Users must use the ``getVal()`` and ``getLoc()``
+methods to access the reduction results.
+
+In the kernel body lambda expression, a ``ValLoc<T,I>`` must be wrapped in a
+``ValOp``, and passed to the lambda in the same order as the corresponding 
+``RAJA::expt::Reduce`` arguments, e.g. ``ValOp<ValLoc<T,I>, minimum>``.
+For convenience, the alias of ``RAJA::expt::ValLocOp<T,I,Op>`` can be used.
```

Comment:
```suggestion
For convenience, an alias of ``RAJA::expt::ValLocOp<T,I,Op>`` can be made.
```

Reply:
Ok, I've written a macro for the users, so I'll reword it to say that it's provided already by us.

Reply:
oh I see! I missunderstood